eth/stagedsync/stage_execute.go

```diff
@@ -418,6 +419,35 @@ func SpawnExecuteBlocksStage(s *StageState, u Unwinder, txc wrap.TxContainer, to
 		logger.Info(fmt.Sprintf("[%s] Blocks execution", logPrefix), "from", s.BlockNumber, "to", to)
 	}
 
+	if useInternalTx && cfg.silkworm != nil {
+		log.Info("Using Silkworm to commit full range", "fromBlock", s.BlockNumber+1, "toBlock", to)
+
+		//  update plain state version so the cache can be notified that the state has moved on
+		_, err = rawdb.IncrementStateVersion(txc.Tx)
+		if err != nil {
+			return fmt.Errorf("writing plain state version: %w", err)
+		}
+		txc.Tx.Commit()
+		txc.Tx = nil
+
+		if lastExecutedBlock, err := silkworm.ExecuteBlocksPerpetual(cfg.silkworm, cfg.db, cfg.chainConfig.ChainID, s.BlockNumber+1, to, uint64(cfg.batchSize), true, true, true); err != nil {
+			if errors.Is(err, silkworm.ErrInterrupted) {
+				logger.Warn(fmt.Sprintf("[%s] Execution interrupted", logPrefix), "lastExecutedBlock", lastExecutedBlock, "err", err)
+				// Remount the termination signal
+				p, err := os.FindProcess(os.Getpid())
+				if err != nil {
+					return err
+				}
+				p.Signal(os.Interrupt)
+				return nil
+			}
+
+			return err
+		}
+
+		return nil
+	}
+
```

Comment:
I understand why you're trying to encapsulate this call into Silkworm `execute_blocks_perpetual` here, but there are some non-trivial implications to take into account, e.g.:
- `IncrementStateVersion` is called before execution, while it is called after in pure Erigon mode
- `UnwindTo` needs to be called when something goes wrong during block execution

I suggest to do things in steps and first of all migrate from single API `silkworm_execute_blocks` to double API `execute_blocks_perpetual` and `execute_blocks_ephemeral` without adding any other change except this one.